opt.d/build-ubuntu-22.04.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/sh -x
+
+# platform-independent lib.sh
+cd `dirname $0` || exit 1
+opt_dir=`pwd`
+. ./lib.sh || exit 1
+
+# platform dependencies
+shared_deps="
+build-essential
+curl
+dirmngr
+libevent-dev
+libpcre3-dev
+libssl-dev
+make
+perl
+zlib1g-dev
+liblzma-dev 
+libzstd-dev   
+pkg-config 
```

Comment:
👀 New dependencies (vs [ubunty-20.04](https://github.com/guardian/eotk/blob/master/opt.d/build-ubuntu-20.04.sh))
```
liblzma-dev 
libzstd-dev   
pkg-config 
```

We're also removing `libssl1.1` as an explicit dependency